Rename Federation V2 to KubeFed in all docs

dfarrell07 opened the pull request:
The Federation project has been renamed KubeFed. This change renames all
appropriate occurrences in the docs.

The format is KubeFed, in UpperCamel, per agreement on 2019-05-15
Federation WG meeting. The longform is Kubernetes Cluster Federation,
per clarification on PR #912. The kubefed2 CLI tool was renamed
kubefedctl, per the 0.0.10 changelog.

Specifically searched for:
* Kubefed (case sensitive)
  * Should not occur except in immutable CHANGELOG.
* kubefed (case sensitive)
  * Should only occur when in code, URLs, Markdown links, similar.
* kubefed (case insensitive)
  * Reviewed for any unexpected capitalizations, found none.
* v2 (case insensitive)
  * Should not occur in relation to Federation
* Federation (case sensitive)
  * Should only occur in immutable CHANGELOG, in reference to history of
    Federation v1, or at the start of a sentence or in heading when
    referring to the concept of federation, not the project name
    "Federation".
* kubefed2
  * Should only occur in immutable CHANGELOG.

Announcement of name change:

https://groups.google.com/d/msg/kubernetes-sig-multicluster
/HHyqrfFHQiw/HJ3F6dOHBAAJ

Signed-off-by: Daniel Farrell <dfarrell@redhat.com>

marun commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
-## Learn About Kubernetes Cluster Federation
-If you want to learn about Kubernetes Cluster Federation, please read through the doc:
-https://github.com/kubernetes-sigs/kubefed#kubernetes-cluster-federation
+## Learn About KubeFed
```
Please revert

dfarrell07 replied:
Done

marun commented on README.md:
```diff
@@ -2,12 +2,12 @@
-[![Releases](https://img.shields.io/badge/version-v0.0.10-orange.svg)](https://github.com/kubernetes-sigs/kubefed/releases "Kubefed latest release")
+[![Releases](https://img.shields.io/badge/version-v0.0.10-orange.svg)](https://github.com/kubernetes-sigs/kubefed/releases "KubeFed latest release")
 
-# Kubernetes Cluster Federation
```
Please revert

dfarrell07 replied:
Done

marun commented on charts/kubefed/README.md:
```diff
@@ -2,17 +2,17 @@
 **Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
 
-- [Kubernetes Cluster Federation](#kubernetes-cluster-federation)
+- [KubeFed](#KubeFed)
```
Please revert

dfarrell07 replied:
Done

marun commented on charts/kubefed/README.md:
```diff
@@ -2,17 +2,17 @@
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-# Kubernetes Cluster Federation
```
Please revert

dfarrell07 replied:
Done

marun commented on charts/kubefed/README.md:
```diff
@@ -2,17 +2,17 @@
+# KubeFed
 
-Kubernetes Cluster Federation is a Kubernetes Incubator project. It builds on the sync controller
+KubeFed is a Kubernetes Incubator project. It builds on the sync controller
```
Please revert

dfarrell07 replied:
Done

marun commented on docs/concepts.md:
```diff
@@ -2,19 +2,19 @@
 | Federate             | Federating a set of Kubernetes clusters means, effectively creating a common interface to the pool of these clusters which can be used to deploy Kubernetes applications across those clusters.                                                                                                                     |
-| Federation           | Kubernetes Cluster Federation enables users to federate multiple Kubernetes clusters for resources distribution, service discovery, high availability etc across multiple clusters.                                                                                                                                 |
-| Host Cluster         | A cluster which is used to expose the federation API and run the kubefed control plane.                                                                                                                                                                                                                          |
+| KubeFed           | KubeFed enables users to federate multiple Kubernetes clusters for resources distribution, service discovery, high availability etc across multiple clusters.                                                                                                                                 |
```
Please revert replacement in second column

dfarrell07 replied:
Done

marun commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
 https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md
 
-If you are interested in contributing to Kubernetes Cluster Federation project, please read through the Development Guide:
+If you are interested in contributing to KubeFed, please read through the Development Guide:
```
Please revert

dfarrell07 replied:
Done, but I removed "project" in the latest version. If we wanted to keep project, I think I'd need to add "the" to make it "the Kubernetes Cluster Federation project".

marun commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
+https://github.com/kubernetes-sigs/kubefed#kubefed
 
-To learn how to use Kubernetes Cluster Federation, please read through the User Guide:
+To learn how to use KubeFed, please read through the User Guide:
```
Please revert

dfarrell07 replied:
Done

dfarrell07 commented on docs/ingressdns-with-externaldns.md:
```diff
@@ -22,7 +22,7 @@ external DNS records of Ingress resources in supported DNS providers.
 
 The above diagram illustrates MCIDNS. A typical MCIDNS workflow consists of:
 
-1. Creating `FederatedDeployment`, `FederatedService`, and `FederatedIngress` resources. The Federation sync
```
This resource name shouldn't have changed. Fixed in latest patch.

dfarrell07 commented on README.md:
```diff
@@ -2,12 +2,12 @@
+# KubeFed
 
 This repo contains an in-progress prototype of some of the
-foundational aspects of V2 of Kubernetes Federation.  The prototype
```
I changed this back to the longform version in the latest patch.

dfarrell07 commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
 ## Contributing Guidelines
-This repo contains an alpha release of some of the foundational aspects of V2 of Kubernetes Federation. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
+This repo contains an alpha release of some of the foundational aspects of KubeFed. All the contribution is done via GitHub. Please also read the contributors guide: https://github.com/kubernetes/community/blob/master/contributors/guide/README.md
```
I changed this back to the longform version in the latest patch.

Jaanki commented:
There are 2 references of kubefed2 in user guide too at below places
1. https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#join-clusters
2. https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#unjoining-clusters

Please correct those too in this PR or another.

dfarrell07 commented:
> There are 2 references of kubefed2 in user guide too at below places
> 
> 1. https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#join-clusters
> 2. https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md#unjoining-clusters
> 
> Please correct those too in this PR or another.

Thanks Janki! Fixed in the latest patch and updated commit message to reflect slight PR scope change.

marun commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
 ## Learn About Kubernetes Cluster Federation
-If you want to learn about Kubernetes Cluster Federation, please read through the doc:
-https://github.com/kubernetes-sigs/kubefed#kubernetes-cluster-federation
+If you want to learn about KubeFed, please read through the doc:
```
Please revert as per a previous comment - any reference to 'Kuberentes Cluster Federation' should remain unchanged.

dfarrell07 replied:
Yeah, I left this one on purpose. I thought it was nice to have the full name, then the short name, to introduce both. Either way is pretty good I think.

marun replied:
Fine to introduce the name, but changing the readme link resulted in a broken link.

dfarrell07 replied:
Oh I see, right. Good catch. :)

marun commented on CONTRIBUTING.md:
```diff
@@ -1,17 +1,17 @@
-If you want to learn about Kubernetes Cluster Federation, please read through the doc:
-https://github.com/kubernetes-sigs/kubefed#kubernetes-cluster-federation
+If you want to learn about KubeFed, please read through the doc:
+https://github.com/kubernetes-sigs/kubefed#kubefed
```
ditto

marun commented:
Thanks for doing this!  I'll submit another PR to revert that one change.

/approve
/lgtm
